src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// QueryManagerFactory. Implements IQueryManagerFactory
+    /// Used to get the appropriate query builder, query executor and exception parser and  based on the database type.
+    /// </summary>
+    public class QueryManagerFactory : IQueryManagerFactory
+    {
+        private readonly IDictionary<DatabaseType, IQueryBuilder> _queryBuilders;
+        private readonly IDictionary<DatabaseType, IQueryExecutor> _queryExecutors;
+        private readonly IDictionary<DatabaseType, DbExceptionParser> _dbExceptionsParsers;
+
+        /// <summary>
+        /// Initiates an instance of QueryManagerFactory
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeconfigprovider.</param>
+        /// <param name="logger">logger.</param>
+        /// <param name="contextAccessor">httpcontextaccessor.</param>
+        public QueryManagerFactory(RuntimeConfigProvider runtimeConfigProvider, ILogger<IQueryExecutor> logger, IHttpContextAccessor contextAccessor)
+        {
+            _queryBuilders = new Dictionary<DatabaseType, IQueryBuilder>();
+            _queryExecutors = new Dictionary<DatabaseType, IQueryExecutor>();
+            _dbExceptionsParsers = new Dictionary<DatabaseType, DbExceptionParser>();
+            foreach (DataSource dataSource in runtimeConfigProvider.GetConfig().ListAllDataSources())
+            {
+                IQueryBuilder? queryBuilder = null;
+                IQueryExecutor? queryExecutor = null;
+                DbExceptionParser? exceptionParser = null;
+
+                if (_queryBuilders.ContainsKey(dataSource.DatabaseType))
+                {
+                    // we have already created the builder, parser and executor for this database type. no need to create again.
+                    continue;
+                }
+
+                switch (dataSource.DatabaseType)
+                {
+                    case DatabaseType.CosmosDB_NoSQL:
```

Comment:
Is it possible to get `CosmosDB_NoSql` here?

Reply:
Yup it is, because you are going over all the datasources.